### PR TITLE
chore(components): update spacing tokens

### PR DESCRIPTION
**Pull request by asudoh**
This PR attempts to migrate away from `v9` spacing tokens by replacing the references with `v10` spacing tokens. And upgrade `carbon-elements`.

Refs #1501.

#### Changelog

**Changed**

- Updated `carbon-elements` dependency.
- Replaced `v9` spacing tokens with `v10` spacing tokens.
- Replaced IBM color variable references to catch up with latest `carbon-elements`.

#### Testing / Reviewing

Testing should make sure no component is broken.

### CODE REVIEW

File: src/components/dropdown/_dropdown.scss [hunk 1]
```diff
@@ -497,7 +497,7 @@
 
   .#{$prefix}--dropdown--inline .#{$prefix}--dropdown-text {
     display: inline-block;
-    padding: rem(7px) rem(32px) rem(7px) $spacing-sm; // TODO: spacing token
+    padding: rem(7px) rem(32px) rem(7px) $carbon--spacing-04; // TODO: spacing token
```
Comment (emyarod) [inline src/components/dropdown/_dropdown.scss:500]: I think we can remove the comment here right?

File: src/globals/scss/vendor/@carbon/elements/scss/grid/12.scss [hunk 2]
```diff
@@ -5,47 +5,33 @@
...
-    ),
-    xlg: (
-      columns: 12,
-    ),
-    max: (
-      columns: 12,
-    ),
-  ),
-  $carbon--grid-breakpoints
+    lg:
```
Comment (emyarod) [inline src/globals/scss/vendor/@carbon/elements/scss/grid/12.scss]: I'm a bit out of the loop on the grid changes, have the `sm` and `md` breakpoints moved?

**Review by emyarod — APPROVED** [review]
all components look good on my end, thanks @asudoh !

File: src/components/code-snippet/_code-snippet.scss [hunk 3]
```diff
@@ -353,16 +353,16 @@
     border: none;
     max-width: rem(760px);
     min-width: rem(320px);
-    height: $spacing-2xl;
-    padding: 0 56px 0 $spacing-md;
+    height: carbon--mini-units(3.5);
```
Comment (alisonjoseph) [inline src/components/code-snippet/_code-snippet.scss]: Do we have a rule when to use mini-units vs when to just use height?

File: src/components/search/_search.scss [hunk 4]
```diff
@@ -188,7 +188,7 @@
     border: none;
     background-color: $field-01;
     color: $text-01;
-    padding: 0 $spacing-2xl;
+    padding: 0 carbon--mini-units(3.5);
```
Comment (alisonjoseph) [inline src/components/search/_search.scss]: Should this be a spacing token because its padding?

File: src/components/text-area/_text-area.scss [hunk 5]
```diff
@@ -146,7 +146,7 @@
   }
 
   .#{$prefix}--text-area--invalid {
-    padding-right: $spacing-2xl;
+    padding-right: carbon--mini-units(3.5);
```
Comment (alisonjoseph) [inline src/components/text-area/_text-area.scss]: same here, should it be spacing token?

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Overall looks good, just had a few small questions 😄

**Comment by joshblack** [comment]
We could use the aliases if we wanted to! e.g `spacing-01` instead of `carbon--spacing-01` 👍

**Comment by asudoh** [comment]
@joshblack The mini-unit usage @alisonjoseph pointed out is due to there is no spacing token associated with `$spacing-2xl` (`carbon--mini-units(3.5)`). Would you think of any better migration path...? Thanks!

**Comment by alisonjoseph** [comment]
Oh, didn't realize there wasn't a spacing token  🤔

**Comment by shixiedesign** [comment]
This looks great! Thanks!

**Review by shixiedesign — APPROVED** [review]
This looks great! Thanks!

**Comment by asudoh** [comment]
Thanks @joshblack for your input offline! Updated upon [the new migration map](https://github.com/IBM/carbon-elements/blob/v0.0.1-beta.1/docs/migration/10.x-layout.md#migrating).

**Comment by asudoh** [comment]
Merging as @alisonjoseph's comment has been addressed. But please don't hesitate to speak up you see any concerns!

**Pull request merged**